### PR TITLE
These tests should NOT fail

### PR DESCRIPTION
**Pull request by ilaflott**
## Describe your changes
split up cli tests, add a new one based on the `catalogbuilder` interface, so we'll know if it changes again via a failed test. 

## Issue ticket number and link (if applicable)
#149 

## Checklist before requesting a review

- [x] I ran my code
- [x] I tried to make my code readable
- [x] I tried to comment my code
- [x] I wrote a new test, if applicable
N/A ~~- [x] I wrote new instructions/documentation, if applicable~~
- [x] I ran pytest and inspected it's output
- [x] I ran pylint and attempted to implement some of it's feedback

### CODE REVIEW

**Comment by ilaflott** [comment]
this PR now contains autogenerated documentation from python doc-strings!
https://github.com/NOAA-GFDL/fre-cli/actions/runs/10372226899/job/28714459606?pr=151

**Comment by ilaflott** [comment]
ready to go

File: fre/app/mask_atmos_plevel.py [hunk 1]
```diff
@@ -49,61 +51,59 @@ def maskAtmosPlevel_subtool(infile, outfile, psfile):
...
     # transpose dims like the original array
     masked = masked.transpose(*ds[var].dims)
-
     print(f"Processed {var}")
 
     return masked
 
 
-def pressure_coordinate(ds, varname, verbose=False):
+def pressure_coordinate(ds, varname):#, verbose=False):
```
Comment (ceblanton) [inline fre/app/mask_atmos_plevel.py:96]: `verbose` wasn't used, I see

File: fre/catalog/frecatalog.py [hunk 2]
```diff
@@ -1,15 +1,20 @@
+'''
+entry point for fre catalog subcommands
+'''
+
 import click
-import catalogbuilder
+#import catalogbuilder
```
Comment (ceblanton) [inline fre/catalog/frecatalog.py:6]: Isn't this needed?

Comment (ilaflott) [reply fre/catalog/frecatalog.py:6]: it is not! if it were, the new API tests in `fre/catalog/tests` would be failing

File: fre/catalog/frecatalog.py [hunk 3]
```diff
@@ -25,17 +30,20 @@ def catalogCli():
 @click.pass_context
 def builder(context, input_path = None, output_path = None, config = None, filter_realm = None,
             filter_freq = None, filter_chunk = None, overwrite = False, append = False):
+    # pylint: disable=unused-argument
     """ - Generate .csv and .json files for catalog """
-    context.forward(gen_intake_gfdl.main)
+    context.forward(gen_intake_gfdl.create_catalog_cli)
```
Comment (ceblanton) [inline fre/catalog/frecatalog.py:35]: this is the update!

File: fre/catalog/frecatalog.py (hunk 2 above)
Comment (ceblanton) [reply fre/catalog/frecatalog.py:6]: I can't disagree with you. But it's weird still isn't it? should we remove the line then?

File: fre/app/mask_atmos_plevel.py (hunk 1 above)
Comment (ilaflott) [reply fre/app/mask_atmos_plevel.py:96]: true! commented out... since it's not a BAD idea

File: fre/catalog/frecatalog.py (hunk 3 above)
Comment (ilaflott) [reply fre/catalog/frecatalog.py:35]: and and and and 

https://github.com/NOAA-GFDL/fre-cli/blob/d237b690cdd4e593cbdea2202ea2ec7578baba47/fre/tests/test_fre_catalog_cli.py

**Review by ceblanton — APPROVED** [review]
Yeah! Lots of bonus good in this one, as well as the intended one line fix for the catalog builder api change.

**Pull request merged**